### DISCUSSION TITLE
Paperless installation, what to do after the helper script

### DISCUSSION THREAD

**Discussion by shgmongo** [tteck/Proxmox#3098] [Help Needed] [answered]
Hi,

I have installed paperless with the[ helper skript](https://helper-scripts.com/scripts?id=Paperless-ngx) on my proxmox machine.

I have access to the the webui from paperless and everthing is working so far.

But how can I create e.g. a consume folder, how can I get ftp access to this folder, how can I change settings of paperless in the console of paperless?

Regards, 
André

**Comment by andreklug** [comment]
Hi André,

you could start by looking at the contents of paperless.conf, where you'll find this: 

`PAPERLESS_CONSUMPTION_DIR=/media/consume/
`
Test if this works, and then change the path or mount it to where your incoming files will be.
Also check the other options in the config file and adapt them to your needs. 

If the basics work, you'll have to think about a meaningful structure for your document types, paths and tags. As an example, here is what I am using for document types:
- Angebot (offer)
- Bestellung (order)
- Anleitungen (documentation)
- Gehaltsabrechnungen (pay slip)
- Gutschein / Gutschrift (voucher / credit note)
- Information 
- Kontoauszug (account statement)
- Kündigung (cancellation)
- Lieferschein (delivery slip)
- Mahnung (reminder)
- Nachfrage / Aufforderung (demand / request)
- Notiz (note)
- Quittung / Beleg (receipt / voucher)
- Rechnung (bill)
- Steuerbescheid / - bescheinigung (tax assessment notice / certificate)
- Urkunde / Zertifikat / Zeugnis / Dokument (Deed / certificate / testimonial / document)
- Vertrag / Vereinbarung (contract / agreement)

I am using tags to mark context: which person is affected (me, wife, kids) or special items (car, house) or if it covers specific topics (health, education, insurance). And I have a tag for outgoing correspondence.


Also, you'll want to give the documents meaningful names. This is what I am using: 

`PAPERLESS_FILENAME_FORMAT={created_year}.{created_month}.{created_day} - {correspondent} - {title}`

If yoiu intend to import a lot of old documents, I would recommend you to not import all at one, but only batches of 10 or 20. make sure they are all properly tagged, and then use the terminal to run the following command (probably in /opt/paperless/)

`python3 manage.py document_create_classifier`

This will rebuild the classifier model and make sure the next documents you are importing already follow what the system has learned from your tagging. 

Hope this helps for a good start!

**Reply by shgmongo** [reply]
Hi, thanks for your help. But I think one of my problems is, how can I open `paperless.conf`? What have I to write into the console to open it.

**Reply by andreklug** [reply]
The easiest way is going directly though the Proxmox web UI:

![grafik](https://github.com/tteck/Proxmox/assets/13343357/6c180322-e7ea-40df-976a-448cf471e639)

**Reply by andreklug** [reply, marked as answer]
when you are in the console, type: 

`find / -name paperless.conf`

then edit the file using nano.

```
root@paper:/# find / -name paperless.conf
/opt/paperless/paperless.conf
nano /opt/paperless/paperless.conf
```

Not sure about your level of experience with Proxmox / Linux, so let me know if you mean something different.

**Reply by shgmongo** [reply]
Thanks, that is what I was searching for. Now I can edit it on my own....

**Comment by Darklirah** [comment]
Hi there,
another noobie joins ;-D
managed to install and run paperless, webgui works.
Now I am asking me, how to add networkattatches storage to paperless to hold the consume, media... etc. folders
In the.conf it does not change anything - so I changed the paths in the docker file:
`VOLUME ["//OPENMEDIAVAULT/Dateiablage-SSD/paperless/data", \
        "//OPENMEDIAVAULT/Dateiablage-SSD/paperless/media", \
        "//OPENMEDIAVAULT/Dateiablage-SSD/paperless/data/consume", \
        "//OPENMEDIAVAULT/Dateiablage-SSD/paperless/export"]`

but instead conecting to the nas it created these dirs and uses them to store the data

half success could change the dirs destination but not connect to the nas folders:-(

hope you could help

cheers to everyone

**Comment by andreklug** [comment]
1. create the directories (in /media or /mnt)
2. mount the network shares into these directories. check permissions by trying to create a folder an a file in that share from the paperless machine, also the other way around
3. change path in the config file to match the local folders
that's all

**Reply by Darklirah** [reply]
first of all   ty for yout superfast response!
to make things clear:
make the mountpoints and then add the mounts in FSTAB as cfis ?

**Comment by andreklug** [comment]
yes, you are correct. (some / most) unix applications tend to create directories they need to write to but don't exist yet. this is what happened to you, I think.

**Comment by supertevran** [comment]
Hi Guys. Is my understanding correct that the base installation script is not offering a samba share eg. to the consume directory by default? I wanna scan my documents directly from my Brother ADS-1700W into this directory either by smb or ftp or .... Has anyone managed to achieve this and if yes - how to do? Thx.

**Reply by Vakrehus** [reply]
I also use a Brother ADS 1600W and have created a profile in the scanner, which then sends to a Samba share (consume folder).
You have to install Samba by yourself, but it's a quick and easy setup.

`apt-get install samba`

`nano /etc/samba/smb.conf`

Edit the config to your needs. Save and Exit.
Restart Samba with:

`/etc/init.d/samba restart`

That's it.

**Comment by andreklug** [comment]
Yes and why should it? But anyway, just do apt install samba, then edit the /etc/samba/smb.conf to match your needs and don't forget to set the right permissions on the folder you're sharing. You might also need to create a firewall rule if you are using one.

But, from my view this approach is not optimal. As a selfhoster, I assume you have some file server somewhere. Or any other directory which you can make sure to integrate into your **backup strategy**. If so, I would suggest let paperless and the scanner access this directory, and then don't run any other services in the paperless container for the sake of maintainability.

**Reply by supertevran** [reply]
Thx. for your kind reply. Of course i already run a fileserver in my setup on a separate lxc container and i fancy the idea of not "bloating" the paperless lxc by adding additional stuff which is already supplied somewhere else. My current understanding is that the "smart" concept of the consume directory in paperless is that it "eats" the files in there and they are "automagically gone" after processing. Scanning into another directory somewhere else (on another machine) leaves me with the task of cleaning this on a regular basis.... Even the documentation advises to supply some kind of service like samba or ftp/sftp. And another basic question related to this issue. Where do i have to "add" smb or "ftpd". In Docker's configuration file? In the lxc-container?

**Reply by andreklug** [reply]
no problem :) 

The consume directory works very well using a remote samba share. 

You are right - once a file is consumed, it will be moved away from that share and moved to another location. You can avoid that by taking the write permission away, but that's not a good idea, because paperless-ngx will always try to consume the files again, see they are duplicates and fill your error log. 
Looking at my setup, a copy of the file is created in the paperless/media folder.
I am wondering why you mention docker? tteck's installation method runs without docker. Therefore the share will be created in the LXC directly. I am mounting mine simply by creating an entry in /etc/fstab:

![grafik](https://github.com/user-attachments/assets/381be735-37b3-4e21-818e-56c372e349dd)

The first one is for the consume directory, the second one is where the processed files will go (in addition to the copy in paperless/media).

**Reply by supertevran** [reply]
Great! I'll try to mimic that in my setup and will come back on that. No docker in the helper script installation methid? Even better!
